vcd/resource_vcd_lb_app_rule.go

```diff
@@ -0,0 +1,220 @@
+	}
+
+	readLBRule, err := edgeGateway.ReadLBAppRuleByID(d.Id())
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("unable to find load balancer app rule with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBAppRuleData(d, readLBRule)
+}
+
+func resourceVcdLBAppRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	updateLBRuleConfig, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("could not expand load balancer app rule for update: %s", err)
+	}
+
+	updatedLBRule, err := edgeGateway.UpdateLBAppRule(updateLBRuleConfig)
+	if err != nil {
+		return fmt.Errorf("unable to update load balancer app rule with ID %s: %s", d.Id(), err)
+	}
+
+	if err := setLBAppRuleData(d, updatedLBRule); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceVcdLBAppRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	err = edgeGateway.DeleteLBAppRuleByID(d.Id())
+	if err != nil {
+		return fmt.Errorf("error deleting load balancer app rule: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// resourceVcdLBAppRuleImport is responsible for importing the resource.
+// The d.Id() field as being passed from `terraform import _resource_name_ _the_id_string_ requires
+// a name based dot-formatted path to the object to lookup the object and sets the id of object.
+// `terraform import` automatically performs `refresh` operation which loads up all other fields.
+//
+// Example import path (id): org.vdc.edge-gw.existing-app-rule
+func resourceVcdLBAppRuleImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified in such way org.vdc.edge-gw.existing-app-rule")
+	}
+	orgName, vdcName, edgeName, appProfileName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	edgeGateway, err := vcdClient.GetEdgeGateway(orgName, vdcName, edgeName)
+	if err != nil {
+		return nil, fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBRule, err := edgeGateway.ReadLBAppRuleByName(appProfileName)
+	if err != nil {
+		return []*schema.ResourceData{}, fmt.Errorf("unable to find load balancer app rule with name %s: %s",
+			d.Id(), err)
+	}
+
+	d.Set("org", orgName)
+	d.Set("vdc", vdcName)
+	d.Set("edge_gateway", edgeName)
+	d.Set("name", appProfileName)
+
+	d.SetId(readLBRule.ID)
```

Comment:
Why for Rule `appProfileName' is saved and why script isn't saved ?

Reply:
Because as we spoke before we only care to get the ID in `import` (other fields org, vdc, edge_gateway and name are required anyway. Could even avoid setting them. Import automatically triggers `refresh` and it fills in the missing pieces. (script in this case)

Reply:
Why appProfleName and not RuleName ?

Reply:
Another one. And I thought I have read it a few times after changing.
Changed. Thanks